ui/admin/tests/e2e/tests/login.spec.js

```diff
@@ -0,0 +1,74 @@
+/* eslint-disable no-undef */
+const { test, expect } = require('@playwright/test');
+
+test('Log in, log out, and then log back in', async ({ page }) => {
+  await page.goto('/');
+
+  await page
+    .getByLabel('Login Name')
+    .fill(process.env.E2E_PASSWORD_ADMIN_LOGIN_NAME);
+  await page
+    .getByLabel('Password')
+    .fill(process.env.E2E_PASSWORD_ADMIN_PASSWORD);
+  await page.getByRole('button', { name: 'Sign In' }).click();
+  await expect(page.getByRole('navigation', { name: 'General' })).toBeVisible();
+  await expect(
+    page.getByText(process.env.E2E_PASSWORD_ADMIN_LOGIN_NAME)
+  ).toBeEnabled();
+
+  await page.getByText(process.env.E2E_PASSWORD_ADMIN_LOGIN_NAME).click();
+  await page.getByRole('button', { name: 'Sign Out' }).click();
+  await expect(page.getByRole('button', { name: 'Sign In' })).toBeVisible();
+
```

Comment:
Is there anything else we can check to make sure we're logged out?

Reply:
I am not sure. The only other thing I can think of (which is something I can see as a real user interaction) is to try to directly change the URL, wait for it to resolve, and confirm that we still see the "Sign In" button. Not sure if that is possible or not and is likely overkill for this test.